Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm

```diff
@@ -993,6 +1049,75 @@ - (void)_processMediaInfoDictionaries:(NSArray *)infos atIndex:(NSUInteger)index
     [self _uploadItemFromMediaInfo:info successBlock:uploadItemSuccessBlock failureBlock:failureBlock];
 }
 
+- (void)_processPickerResults:(NSArray<PHPickerResult *> *)results atIndex:(NSUInteger)index processedResults:(NSMutableArray<_WKFileUploadItem *> *)processedResults successBlock:(void (^)(NSArray<_WKFileUploadItem *> *processedResults))successBlock failureBlock:(void (^)(void))failureBlock
+{
+    NSUInteger count = [results count];
+    if (index == count) {
+        successBlock(processedResults);
+        return;
+    }
+
+    PHPickerResult *result = [results objectAtIndex:index];
+    ASSERT(index < count);
+    index++;
+
+    auto uploadItemSuccessBlock = ^(_WKFileUploadItem *uploadItem) {
+        [processedResults addObject:uploadItem];
+        [self _processPickerResults:results atIndex:index processedResults:processedResults successBlock:successBlock failureBlock:failureBlock];
+    };
+
+    [self _uploadItemFromResult:result successBlock:uploadItemSuccessBlock failureBlock:failureBlock];
+}
+
+- (void)_uploadItemFromResult:(PHPickerResult *)result successBlock:(void (^)(_WKFileUploadItem *))successBlock failureBlock:(void (^)(void))failureBlock
+{
+    if (NSString *uti = firstUTIThatConformsTo(result.itemProvider.registeredTypeIdentifiers, UTTypeMovie)) {
+        [result.itemProvider loadFileRepresentationForTypeIdentifier:uti completionHandler:^(NSURL *url, NSError *error) {
```

Comment:
We will need to copy these files, as they will be destroyed by the time the data is returned to the web process.